core/src/Streamly/Internal/Data/Array/Generic.hs

```diff
@@ -118,9 +118,9 @@ createOf = fmap unsafeFreeze <$> MArray.createOf
 writeN :: MonadIO m => Int -> Fold m a (Array a)
 writeN = createOf
 
-{-# INLINE_NORMAL writeWith #-}
-writeWith :: MonadIO m => Int -> Fold m a (Array a)
-writeWith elemCount = unsafeFreeze <$> MArray.createWith elemCount
+{-# INLINE_NORMAL createWith #-}
+createWith :: MonadIO m => Int -> Fold m a (Array a)
+createWith elemCount = unsafeFreeze <$> MArray.createWith elemCount
```

Comment:
How is this different from createOf? if it is different we should document it.

Reply:
`create = createWith (allocBytesToElemCount (undefined :: a) arrayChunkBytes)`

Reply:
`createWith` is documented

Reply:
Ah, it's documented in the `MutArray` module.